### ISSUE TITLE
consider a different CI provider for iOS?

### ISSUE THREAD

**Issue by tomholub** [FlowCrypt/flowcrypt-ios#518]
We use SemaphoreCI on all projects, but here for iOS UI tests it was giving us trouble - tests would run successfully on local mac Xcode but not run successfully on CI.

It's hard to say if it's our code or SemaphoreCI at fault without investigating further - so I'll ask Dmitry to look into it.

It may turn out that the issues may be solved by a combination of:
 - adjusting the code to run well in CI
 - using another CI provider
 - using Appium

If it does turn out we should try another CI provider, then here are some options I've seen so far:
- https://www.bitrise.io/why/technologies/ios-continuous-integration (Dmitry mentioned issue with secrets?)
- https://developer.apple.com/documentation/Xcode/About-Continuous-Integration-and-Delivery-with-Xcode-Cloud
- https://circleci.com/docs/2.0/testing-ios/
- https://codemagic.io/ios-continuous-integration/
- GitLab also has CI for iOS but I suppose the GitHub integration may not be very good? https://about.gitlab.com/blog/2016/03/10/setting-up-gitlab-ci-for-ios-projects/

**Comment by tomholub** [comment]
> for semaphoreci can I create new project? cause I feel like I can break existing with my changes
> so i can copy ios project and test

There's nothing to break if you work on your own branch then you make a PR and I review it, just like all other changes.

You'll want to add a job here: https://github.com/FlowCrypt/flowcrypt-ios/blob/master/.semaphore/semaphore.yml with the commands to run to install and run appium.

The only change on the actual semaphore project is to add a secrets file. That's not a change that would break anything.

**Comment by tomholub** [comment]
Don't put the password in the code, that's what semaphore secrets are for. I'll change the test account password now and distribute it over email again.

**Comment by tomholub** [comment]
Doesn't matter where it is. If I can read it on git, then it's in the wrong place.

![image](https://user-images.githubusercontent.com/6306961/138051069-46d4031f-00c0-41aa-a74d-3c1093dfcec7.png)

**Comment by tomholub** [comment]
You should edit that file directly by hand. Don't use the visual thing on semaphore.

I've rotated the password, sent a new env file. I've also configured a secret on Semaphore so that you will have that env file in `/Users/semaphore/appium-env` - you can move it wherever you want to during setup.

**Comment by tomholub** [comment]
Also, load latest changes from master, I've edited the semaphore file since.

**Comment by fcvakintos** [comment]
got you

**Comment by fcvakintos** [comment]
@tomholub tests can run, at least on ci. It's a screenshot from execution
<img width="345" alt="Screenshot 2021-10-20 at 11 36 02" src="https://user-images.githubusercontent.com/91205853/138058375-029772e1-e05d-457f-b697-8d1d40790d40.png">

**Comment by fcvakintos** [comment]
@tomholub appium-env how to check what is inside it?

**Comment by tomholub** [comment]
Yes, you'll want to adjust the previous build step to do that (or create a new one).

**Comment by tomholub** [comment]
I'll have a look. I've made a draft PR because it's easier to discuss it that way.
